tcms/rpc/tests/test_testcase.py

```diff
@@ -3,26 +3,86 @@
 from tcms_api import xmlrpc
 
 from tcms.core.helpers import comments
 from tcms.management.models import Priority
+from tcms.rpc.api.testcase import _validate_cc_list
 from tcms.rpc.tests.utils import APITestCase, APIPermissionsTestCase
+from tcms.rpc.utils import add_attachment
 from tcms.testcases.models import Category, TestCase, TestCaseStatus
 from tcms.tests import remove_perm_from_user
 from tcms.tests.factories import (CategoryFactory, ComponentFactory,
                                   ProductFactory, TagFactory, TestCaseFactory,
                                   TestPlanFactory, UserFactory, VersionFactory)
 
 
-class TestNotificationRemoveCC(APITestCase):
-    """ Tests the XML-RPC testcase.notication_remove_cc method """
+class TestValidateEmail(DjangoBaseTestCase):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.default_cc = 'example@example.com'
+
+    def test_non_list_email_collection(self):
+        with self.assertRaisesRegex(TypeError, 'cc_list should be a list object.'):
+            _validate_cc_list((self.default_cc,))
+
+    def test_invalid_email(self):
+        with self.assertRaises(ValidationError):
+            _validate_cc_list(['@example.com'])
+
+
+class TestAddNotificationCCPermission(APIPermissionsTestCase):
+    permission_label = 'testcases.change_testcase'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.default_cc = 'example@example.com'
+        self.testcase = TestCaseFactory()
+
+    def verify_api_with_permission(self):
+        self.rpc_client.TestCase.add_notification_cc(self.testcase.pk, [self.default_cc])
+        self.assertEqual(len(self.testcase.emailing.get_cc_list()), 1)
+        self.assertEqual(self.testcase.emailing.get_cc_list(), [self.default_cc])
+
+    def verify_api_without_permission(self):
+        with self.assertRaisesRegex(ProtocolError, '403 Forbidden'):
+            self.rpc_client.TestCase.add_notification_cc(self.testcase.pk, [self.default_cc])
+
+
+class TestGetNotificationCCPermission(APIPermissionsTestCase):
+    """ Tests the XML-RPC Testcase.get_notification_cc method """
+
+    permission_label = 'testcases.view_testcase'
 
     def _fixture_setup(self):
         super()._fixture_setup()
 
-        self.default_cc = 'example@MrSenko.com'
+        self.default_cc = 'example@example.com'
+        self.testcase = TestCaseFactory()
+        self.testcase.emailing.add_cc(self.default_cc)
+
+    def verify_api_with_permission(self):
+        cc_list = self.rpc_client.TestCase.get_notification_cc(self.testcase.pk)
+        self.assertListEqual(cc_list, [self.default_cc])
+
+    def verify_api_without_permission(self):
+        with self.assertRaisesRegex(ProtocolError, '403 Forbidden'):
+            self.rpc_client.TestCase.get_notification_cc(self.testcase.pk)
+
+
+class TestRemoveNotificationCC(APITestCase):
+    """ Tests the XML-RPC testcase.remove_notification_cc method """
```

Comment:
API method names are case sensitive so if you are going to document them then use the same case as in the method doc-string. Also makes searching for things easier.